python_modules/dagster/dagster/core/asset_defs/assets_job.py

```diff
@@ -127,7 +136,26 @@ def asset2(asset1):
         if not source_asset.io_manager_def:
             required_io_manager_keys.add(source_asset.get_io_manager_key())
         else:
+            # Explicitly check top-level dependencies so we can give error
```

Comment:
This strikes me as a little awkward. Would we not want to give the further transitive errors in terms of the asset key as well?

Reply:
Was thinking about that. I think the thing we don't like about the top-layer error messages is that it references a system-provided key. I see your point about it being a bit awkward though